### PR TITLE
#187419151-Initial Project Setup with TypeScript, PostgreSQL, and Sequelize

### PR DESCRIPTION
**Pull request by soleil00**
#### What does this PR do?

This PR sets up the initial project structure with TypeScript, PostgreSQL for the database, and Sequelize as the ORM. It includes the creation of a dummy user model in the database and sets up basic server functionality.

#### Description of Task to be completed?

- Initialize project with TypeScript, PostgreSQL, and Sequelize.
- Create folder structure for the project.
- Implement a basic user model in the database.
- Set up a basic home route ("/") to confirm server functionality.

#### How should this be manually tested?

1. Run `npm run migrate` to execute the initial database migration and create the test tables in db.
2. Run `npm run seed` to seed the database with dummy user data.
3. Start the development server using `npm run dev`.
4. Access the home route ("/") in your browser or API testing tool to verify server response.
5. Check the user route ("/api/v1/users") to confirm that dummy user data is accessible.

#### Any background context you want to provide?

This PR is part of the initial project setup. It includes the configuration of essential technologies and the creation of foundational components for further development.

#### What are the relevant pivotal tracker/Trello stories?
#187419151
#### Screenshots 
Screenshot 2024-04-13 at 13 05 37
Screenshot 2024-04-13 at 13 16 46

### CODE REVIEW

**Review by niyobertin — COMMENTED** [review]
Looks good to me. But you need to improve branch naming. eg: adding story ID

**Comment by teerenzo** [comment]
> initialised project with ts with postgress for db and sequelize for ORM added first migration and seeders added home routes added user route to fetch all users in db(checking db config and seeders) added folder structure README not updated

**Comment by teerenzo** [comment]
@soleil00 could your refer to this document of how to name your PR and the PR description format https://github.com/atlp-rwanda/engineering-playbook/wiki/3.-Pull-Request, because I am not going to review this kind of PR

**Review by niyobertin — COMMENTED** [review]
Looking good to me

**Review by Heisjabo — APPROVED** [review]
Looks good

**Review by MugemaneBertin2001 — COMMENTED** [review]
LGTM

File: index.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import sequelize, { connect } from "./src/config/dbConnection";
+import app from "./src/utils/server";
+
+app.listen(5000, async () => {
```
Comment (teerenzo) [inline index.ts]: why hardcoded port

**Review by teerenzo — CHANGES_REQUESTED** [review]
remember also to add .env.example with all variables your are using

File: index.ts (hunk 1 above)
Comment (soleil00) [reply index.ts]: let me fix that

Comment (soleil00) [reply index.ts]: i have squahed commits from ch-eslint-test-#187419154 into ch-project-setup-#187419151 and closed new PR kindly review and merge accordingly , i din't update PR mssage

Comment (soleil00) [reply index.ts]: again once this is merged i can continue , i still have many local chnage that i didn't add for sake of avoid overloading many chnage into this single PR

File: package.json [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    "build": "tsc",
+    "migrate": "npx sequelize-cli db:migrate",
+    "seed": "npx sequelize-cli db:seed:all",
+    "lint": "npx eslint .",
+    "lint:fix": "npx eslint --fix .",
+    "test": " npm run lint && jest --detectOpenHandles --coverage"
+  },
+  "author": "atlp",
+  "license": "MIT",
+  "devDependencies": {
```
Comment (teerenzo) [inline package.json:18]: add this file in .gitIgnore

**Review by teerenzo — CHANGES_REQUESTED** [review]

File: package-lock.json
Comment (teerenzo) [inline package-lock.json:1]: add this in .gitIgnore

**Review by teerenzo — APPROVED** [review]

**Review by istafrica-dennw — CHANGES_REQUESTED** [review]
You have secrets in your codes.

**Pull request merged**